src/cli/self_update/windows.rs

```diff
@@ -511,27 +505,14 @@ fn _apply_new_path(new_path: Option<Vec<u16>>) -> Result<()> {
 // this returns None then the PATH variable is not a string and we
 // should not mess with it.
 fn get_windows_path_var() -> Result<Option<Vec<u16>>> {
-    use std::io;
-
-    let root = RegKey::predef(HKEY_CURRENT_USER);
-    let environment = root
-        .open_subkey_with_flags("Environment", KEY_READ | KEY_WRITE)
+    let environment = CURRENT_USER
+        .create("Environment")
         .context("Failed opening Environment key")?;
 
-    let reg_value = environment.get_raw_value("PATH");
+    let reg_value = environment.get_hstring("PATH");
     match reg_value {
-        Ok(val) => {
-            if let Some(s) = from_winreg_value(&val) {
-                Ok(Some(s))
-            } else {
-                warn!(
-                    "the registry key HKEY_CURRENT_USER\\Environment\\PATH is not a string. \
-                       Not modifying the PATH variable"
-                );
-                Ok(None)
-            }
```

Comment:
This whole `Ok(None)` case has disappeared for no reason.

Reply:
I think `from_winreg_value()` and `to_winreg_bytes()` are useless, and `val` now has type `HSTRING`. Because of that, I removed these statements.

Reply:
> I think `from_winreg_value()` and `to_winreg_bytes()` are useless, and `val` now has type `HSTRING`. Because of that, I removed these statements.

@InfyniteHeap The thing is `PATH` can be of another type other than a string-like one, and you should be able to return `Ok(None)` in this case.